modules/azure-gaming/unreal-cloud-ddc/main.bicep

```diff
@@ -19,12 +19,48 @@ param aksName string = 'ddc-storage-${take(uniqueString(resourceGroup().id), 6)}
 @description('Number of Kubernetes Nodes')
 param agentPoolCount int = 3
 
+@description('Whether to use a local ephemeral Persistent Volume provisioner for the cluster')
+param enableLocalPVProvisioner bool = true
+
+// By default, use one less replica than the nodes in the agent pool if local PV provisioning is enabled.
+// This allows for the ephemeral volume controller to work properly while creating an extra pod during updates.
+// TODO: Make use of all agents even with local PV provisioning.
+@description('Number of pod replicas for the main Kubernetes Deployment')
+param mainReplicaCount int = enableLocalPVProvisioner ? agentPoolCount - 1 : agentPoolCount
+
 @description('Name of Kubernetes Agent Pool')
 param agentPoolName string = 'k8agent'
 
 @description('Virtual Machine Skew for Kubernetes')
 param vmSize string = 'Standard_L16as_v3' // 16 vCPU, 128 GB RAM
 
+@description('Kubernetes version should be supported in all requested regions')
+param kubernetesVersion string = '1.24.9'
```

Comment:
![image](https://github.com/AmieDD/amiedansby-unrealddc/assets/7669428/e864456a-ad2d-4c0c-9d99-b57ad02aed84)
@manuvar-ms can we validate that this version of kubernetes is correct? I get the error that kubernetes version 1.24.9 is not supported in westus3.

Reply:
I've noticed that minimum version keeps changing. We pass '1.25.11' as the `kubernetesVersion` parameter from our Bicep layer. Perhaps we should just remove the default value in this file...?